.circleci/config.yml

```diff
@@ -8,27 +8,34 @@ commands:
         type: string
     steps:
       - checkout
+      - restore_cache:
+          key: -v0-macos-sys-{{ .Branch }}-<< parameters.py_version >>
       - run:
-          name: "Installing Conda"
+          name: "Set up Conda"
           command: |
-            curl -o Miniconda3-latest-MacOSX-x86_64.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
-            bash ./Miniconda3-latest-MacOSX-x86_64.sh -b
+            if [[ -f ~/miniconda3/LICENSE.txt ]] ; then
+              echo "miniconda installed already."
+            else
+              curl -o Miniconda3-latest-MacOSX-x86_64.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+              bash ./Miniconda3-latest-MacOSX-x86_64.sh -b
+            fi
```

Comment:
You will need to switch to a versioned link for miniconda and use it in the cache tag.
right now, if the miniconda link updates we will not upgrade until the cache times out and suddenly we will start using the new one out of the blue.

Reply:
That's a great point! I will update this.

Reply:
I was not able to find a way to ping conda version in windows. For now I updated the MAC OS installation.